src/catalog/database_catalog.cpp

```diff
@@ -550,6 +544,13 @@ bool DatabaseCatalog::DeleteColumns(transaction::TransactionContext *const txn,
         reinterpret_cast<const uint32_t *const>(pr->AccessWithNullCheck(table_pm[ATTNUM_COL_OID]));
     TERRIER_ASSERT(col_name != nullptr, "OID shouldn't be NULL.");
 
+    auto *stored_expression =
+        *reinterpret_cast<parser::AbstractExpression **>(pr->AccessForceNotNull(table_pm[ADBIN_COL_OID]));
+
+    if (stored_expression != nullptr) {
+      txn->RegisterCommitAction([=]() { delete stored_expression; });
```

Comment:
Who cleans `stored_expression` up in the case of an abort? We should probably register a Commit and Abort action.